machineconfiguration/v1alpha1/types_machineconfignode.go

```diff
@@ -0,0 +1,151 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == self.spec.node.name",message="spec.node.name should match metadata.name"
+
+// MachineConfigNode describes the health of the Machines on the system
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type MachineConfigNode struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec describes the configuration of this specific machineconfignode
+	// +kubebuilder:validation:Required
+	Spec MachineConfigNodeSpec `json:"spec"`
+
+	// status describes the last observed state of this machineconfignode
+	// +optional
+	Status MachineConfigNodeStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineConfigNodeList describes all of the MachinesStates on the system
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type MachineConfigNodeList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []MachineConfigNode `json:"items"`
+}
+
+// MCOObjectReference holds information about an object the MCO either owns
+// or modifies in some way
+type MCOObjectReference struct {
+	// name describes what this object is called
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+}
+
+// MachineConfigNodeSpec describes the type of State we are managing
+type MachineConfigNodeSpec struct {
+	// node contains references to the node for this MachineConfigNode
```

Comment:
For consistency with product docs, can we make it `machine config node` in all places we mention it within the godoc? LIkewise machine config pool and others
```suggestion
	// node contains references to the node for this machine config node.
```